drivers/modem/quectel-bg9x.c

```diff
@@ -1077,6 +1079,8 @@ static void modem_net_iface_init(struct net_if *iface)
 			     sizeof(data->mac_addr),
 			     NET_LINK_ETHERNET);
 	data->net_iface = iface;
+
+	iface->if_dev->socket_offload = offload_socket;
```

Comment:
Instead of directly manipulating the pointer in iface, an API function in net_if would be more convenient here.

Reply:
Added API function to set/get the pointer.